src/main/java/io/lettuce/core/models/events/CommandFailedEvent.java

```diff
@@ -0,0 +1,18 @@
+package io.lettuce.core.models.events;
+
+/**
+ * @author sokomishalov
+ */
+public class CommandFailedEvent<K, V, T> extends CommandBaseEvent<K, V, T> {
+
+    private final Throwable cause;
+
+    public CommandFailedEvent(CommandStartedEvent<K, V, T> startedEvent, Throwable cause) {
```

Comment:
Instead of holding to the started event, we should propagate only the context and the command.

Reply:
fixed